Feat(eos_validate_state): ANTA New action plugin to generate the reports (#3352)
aristanetworks/ansible-avd: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md

## Roadmap

!!! note
    Subject to change. No commitments implied.

- Provide a custom ANTA test catalog as a YAML file or by input variables.
- Ability to exclude some interfaces from the report.
- More tests!

## Expected changes

- You should expect faster execution, and if not please report on the GitHub [discussions board](https://github.com/aristanetworks/ansible-avd/discussions)
- Hardware tests are now collapsed.
- Some description of tests have been updated to be more precise.
- Sorting of the test results is now done per device as opposed to per category.
- Tests skipped by ANTA will be marked as `SKIPPED` in the final reports.
- All tests will be removed from the catalog for a device flagged as undeployed using the host level variable [`is_deployed: false`](https://avd.sh/en/stable/roles/eos_designs/docs/input-variables.html#flagging-a-device-as-not-deployed). Additionally, all tests take into account the `is_deployed` variable value and remove tests accordingly.

!!! warning

>     Tests can also be automatically removed from the catalogs depending on the structured configuration of the devices. ANTA is therefore not aware of these tests and they will not appear in the final report. For example, the `AvdTestMLAG` tests will not be present in a test catalog of a device that do not have an MLAG configuration in his structured configuration.

Reviewer: wondering if we should not word it as "the tests are not generated in the catalog, so not forwarded to the test framework (ANTA) and so not in the report".

```suggestion
    Tests can also be automatically removed from the catalogs depending on the structured configuration of the devices. ANTA is therefore not aware of these tests and they will not appear in the final report. For example, the `AvdTestMLAG` tests will not be present in the test catalog of a device that does not have an MLAG configuration in its structured configuration.
```

Author: Fixed in 68479d7a6549495d8a3a4e5ce970b358e51ea5bc

  ansible-playbook playbooks/fabric-validate.yaml --tags routing_table
  ```

- You can now run the eos_validate_state role in check_mode. This will produce a report of tests that will be performed without running the tests on your network. Tests will be marked as `NOT RUN` in the final reports.

  ```shell
  ansible-playbook playbooks/fabric-validate.yaml --check
  ```

> - You can increase the Ansible verbosity by adding multiple `-v` when running the playbook. This will give you visibility on which [test categories](#test-categories) are being removed from the catalogs by AVD according to the structured configurations.

Reviewer: for each device?

Author: Fixed in 2f7c91b74a5914fd1e7fabd3bb86711dcbba55b7

eos_validate_state_name: 'reports'
eos_validate_state_dir: '{{ root_dir }}/{{ eos_validate_state_name }}'

# Output for test results JSON files if save_results is set to true
test_results_dir_name: 'test_results'
test_results_dir: '{{ eos_validate_state_dir }}/{{ test_results_dir_name }}'

# Fabric name used in the reports name
fabric_name: "all"

# Reports name
eos_validate_state_md_report_path: '{{ eos_validate_state_dir }}/{{ fabric_name }}-state.md'
eos_validate_state_csv_report_path: '{{ eos_validate_state_dir }}/{{ fabric_name }}-state.csv'

# Allow different manufacturers

> accepted_xcvr_manufacturers: "{{ validation_role.xcvr_own_manufacturers | arista.avd.default(['Arastra, Inc.', 'Arista Networks']) }}"

Reviewer: is this actually correct?

Reviewer: is it really an input variable or should it be
```
validation_role:
  xcvr_own_manufcatures: [dsjalkdajs]
```

Author: Yes this is what's being used today:
https://avd.sh/en/stable/roles/eos_validate_state/index.html#default-variables
